aiida/backends/sqlalchemy/models/group.py

```diff
@@ -57,6 +59,13 @@ class DbGroup(Base):
     Index('db_dbgroup_dbnodes_dbnode_id_idx', table_groups_nodes.c.dbnode_id)
     Index('db_dbgroup_dbnodes_dbgroup_id_idx', table_groups_nodes.c.dbgroup_id)
 
+    def __init__(self, *args, **kwargs):
+        """Add a default for the extras column."""
+        super().__init__(*args, **kwargs)
+
+        if self.extras is None:
+            self.extras = dict()
+
```

Comment:
I wonder if this is necessary. I know this is what the `Node` class does, but what happens if we remove it? Couldn't our backend layer not just return `{}` if it gets `None`? Also, what is the default value in the database? Is it null? Or should we add `nullable` explicitly to the column declaration?

Note that this is asymmetric with the Django solution. There the default empty dict is set on the column instead of in the logic. We may want to homogenize it.

Reply:
I copied the logic from `DbNode` indeed, but I agree that it makes more sense to simply set a default in the column declaration. I've done so now for both the `DbNode` and `DbGroup` model.

After taking a closer look at the idea of setting the `extras` column to `null` by default in the database, I think it's not really necessary. For nodes, the extras are almost never empty, because [when the node is stored the AiiDA hash is written to the extras](https://github.com/aiidateam/aiida-core/blob/develop/aiida/orm/nodes/node.py#L1075). Groups can of course have empty extras, but usually a user won't have that many groups for it to make a difference, so I don't think it's worth adding extra logic to the backend entities do deal with this.